db/sg_replicate_cfg.go

```diff
@@ -837,6 +837,16 @@ func (m *sgReplicateManager) UpsertReplication(replication *ReplicationUpsertCon
 			replicationConfig.ID = replication.ID
 			cluster.Replications[replication.ID] = &ReplicationCfg{ReplicationConfig: replicationConfig}
 		}
+
+		state, err := m.GetReplicationStatus(replication.ID, DefaultReplicationStatusOptions())
+		errNotFound := errors.Is(err, base.ErrNotFound)
+		if err != nil && !errNotFound {
+			return true, err
+		}
+		if !errNotFound && state.Status != ReplicationStateStopped {
+			return true, base.HTTPErrorf(http.StatusBadRequest, "Replication must be stopped before updating config")
+		}
```

Comment:
I think this can be moved up into the `exists {` block above. We only need to do the status check for replications that already exist, which should avoid the awkward error handling for NotFound.

Reply:
I considered this but we did GetReplicationStatus so it goes to the bucket if need be right? Whereas that exists map wouldn't do that

Reply:
The callback gets the full set of replications in the map, but not all of those are locally running replications, so not as simple as pulling the status out of the replications directly.

We do still need that call to `GetReplicationStatus`, which is smart enough to know whether to go down to the bucket, or retreieve a running local replication directly, but it's only worth doing that if we know a replication exists. If this upsert is for a brand new replication, we don't care about the state.